set.mm

```diff
@@ -609582,6 +609582,66 @@ fixed reference functional determined by this vector (corresponding to
 #*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
 $)
 
+$(
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+  Various arithmetic theorems
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+$)
+
+  $( Alternate version of ~ 1t1e1 using a different set of axioms.
+     (Contributed by Steven Nguyen, 20-Sep-2022.)
+     (Proof modification is discouraged.)  (New usage is discouraged.) $)
+  1t1e1ALT $p |- ( 1 x. 1 ) = 1 $=
+    ( c1 cr wcel cmul co wceq 1re ax-1rid ax-mp ) ABCAADEAFGAHI $.
+
+  ${
+    sqsumi.1 $e |- A e. CC $.
+    sqsumi.2 $e |- B e. CC $.
+    $( A sum squared.  (Contributed by Steven Nguyen, 16-Sep-2022.) $)
+    sqsumi $p |- ( ( A + B ) x. ( A + B ) ) =
+      ( ( ( A x. A ) + ( B x. B ) ) + ( 2 x. ( A x. B ) ) ) $=
+      ( caddc co cmul c2 muladdi mulcli 2timesi eqcomi oveq2i eqtri ) ABEFZOGFA
+      AGFBBGFEFZABGFZQEFZEFPHQGFZEFABABCDCDIRSPESRQABCDJKLMN $.
+  $}
+
+  ${
+    diffsqi.a $e |- A e. CC $.
+    diffsqi.n $e |- N e. CC $.
+    diffsqi.b $e |- ( A + N ) = B $.
+    diffsqi.c $e |- ( B + N ) = C $.
+    $( Difference of squares.  (Contributed by Steven Nguyen, 20-Sep-2022.) $)
+    diffsqi $p |- ( B x. B ) = ( ( A x. C ) + ( N x. N ) ) $=
+      ( caddc co cmul c2 sqsumi mulcli 2cn eqcomi oveq1i 3eqtr3i oveq2i eqtri
+      oveq12i add32i addassi addcli adddii 2timesi mul12i 3eqtrri ) ADIJZUIKJAA
+      KJZDDKJZIJLADKJZKJZIJZBBKJACKJZUKIJZADEFMUIBUIBKGGUAUNUJUMIJZUKIJUPUJUKUM
+      AAEENDDFFNLULOADEFNNUBUQUOUKIUOAADDIJZIJZKJUJAURKJZIJUQCUSAKBDIJUIDIJCUSB
+      UIDIUIBGPQHADDEFFUCRSAAUREEDDFFUDUEUTUMUJIUTALDKJZKJUMURVAAKVAURDFUFPSALD
+      EOFUGTSUHQTR $.
+  $}
+
+  ${
+    sqn5i.1 $e |- A e. NN0 $.
+    $( The square of a number ending in 5.  This shortcut only works because 5
+       is half of 10.  (Contributed by Steven Nguyen, 16-Sep-2022.) $)
+    sqn5i $p |- ( ; A 5 x. ; A 5 ) = ; ; ( A x. ( A + 1 ) ) 2 5 $=
+      ( c5 cdc cmul co cc0 caddc c2 0nn0 deccl nn0cni 5nn0 eqid addid2i decaddi
+      5cn 2nn0 cn0 eqtri c1 5p5e10 decaddci2 diffsqi 5t5e25 peano2nn0 nn0mulcli
+      wcel ax-mp decmulnc mul01i deceq2i 2cn mul02i oveq1i decma ) ACDZUQEFAGDZ
+      AUAHFZGDZEFCCEFZHFAUSEFZIDZCDURUQUTCURAGBJKLQAGCURCBJMURNZCQOZPACUSUQCBMM
+      UQNUSNUBUCUDAGICUTVCCURVABJRMVDUEUSGASUHUSSUHBAUFUIZJKZVBGIAUTEFZIAUSBVFU
+      GJRVHVBAGEFZDVBGDUSGABVFJUJVIGVBAABLUKULTIUMOPGUTEFZCHFGCHFCVJGCHUTUTVGLU
+      NUOVETUPT $.
+
+    sqn5ii.2 $e |- ( A + 1 ) = B $.
+    sqn5ii.3 $e |- ( A x. B ) = C $.
```

Comment:
Your mathbox your choice, but I would suggest replacing these two hypothses with
```
$e |- B = ( ( A x. ( A + 1 ) )  $.
```
(in local definitions, it's usual to put the definiendum on the LHS)

Reply:
The applications I had in mind were multiplying numbers ending in 5. Would the current statements be slower than the combined hypothesis?